### ISSUE TITLE
How to use wasm file in Foxglove extension?

### ISSUE THREAD

**Issue by RaulCHOU** [foxglove/community#868]
### Discussed in https://github.com/orgs/foxglove/discussions/866

<div type='discussions-op-text'>

<sup>Originally posted by **RaulCHOU** October 31, 2023</sup>
**Background**: use C++ source code in Foxglove extension (typescript)
**Method I used**: use "em++" to build my c++ code, then somehow try to use it
**Status now**: I can make it use in my local node.js environment. But failed in extension when start up foxglove
Local valid code:
```
const fs = require('fs');    
const { WASI } = require('wasi');
     
const wasi = new WASI({
    args: process.argv,
    env: process.env,});
const importObject = { 
    env: {
        env: process.env,
    },
    wasi_snapshot_preview1: wasi.wasiImport
 };    
 (async () => {      
const wasm = await WebAssembly.compile(fs.readFileSync('./wrapper.wasm'));     
 const instance = await WebAssembly.instantiate(wasm, importObject);      
 wasi.start(instance);     
})();
```

For extension tpyescript, I used `WebAssembly.instantiate`, emm... I didn't figure it out.
I also use some other method like FFI (webpackage version gap leads to failure )or Node-Addon-Js (cannot load .node file in web environment, right?). 
Do we have any best practice about this usage? I have blocked by it over weeks. 😮‍💨
Thank a lot in advance.</div>

**Comment by jtbandes** [comment]
I'm closing this issue because it's a duplicate of #866 and we don't use Issues in this repo.
